circuits/kimchi/src/gates/endosclmul.rs

```diff
@@ -53,78 +55,48 @@ impl<F: FftField> CircuitGate<F> {
+            gamma: F::zero(),
+            joint_combiner: F::zero(),
+            mds: vec![],
+            endo_coefficient: cs.endo,
+        };
 
-        ensure_eq!(
-            F::zero(),
-            this[11] - this[11].square(),
-            "endomul: wrong eq 1"
-        );
-        ensure_eq!(
-            F::zero(),
-            this[12] - this[12].square(),
-            "endomul: wrong eq 2"
-        );
-        ensure_eq!(
-            F::zero(),
-            this[13] - this[13].square(),
-            "endomul: wrong eq 3"
-        );
-        ensure_eq!(
-            F::zero(),
-            this[14] - this[14].square(),
-            "endomul: wrong eq 4"
-        );
-        ensure_eq!(
-            F::zero(),
-            (xq1 - this[4]) * this[9] - (this[11].double() - F::one()) * this[2] + this[5],
-            "endomul: wrong eq 5"
-        );
-        ensure_eq!(
-            F::zero(),
-            (this[4].double() - this[9].square() + xq1)
-                * ((this[4] - this[7]) * this[9] + this[8] + this[5])
-                - (this[4] - this[7]) * this[5].double(),
-            "endomul: wrong eq 6"
-        );
-        ensure_eq!(
-            F::zero(),
-            (this[8] + this[5]).square()
-                - (this[4] - this[7]).square() * (this[9].square() - xq1 + this[7]),
-            "endomul: wrong eq 7"
-        );
-        ensure_eq!(
-            F::zero(),
-            (xq2 - this[7]) * this[10] - (this[13].double() - F::one()) * this[2] + this[8],
-            "endomul: wrong eq 8"
-        );
-        ensure_eq!(
-            F::zero(),
-            (this[7].double() - this[10].square() + xq2)
-                * ((this[7] - this[2]) * this[10] + this[3] + this[8])
-                - (this[7] - this[2]) * this[8].double(),
-            "endomul: wrong eq 9"
-        );
-        ensure_eq!(
-            F::zero(),
-            (this[3] + this[8]).square()
-                - (this[7] - this[2]).square() * (this[10].square() - xq2 + this[2]),
-            "endomul: wrong eq 10"
-        );
-        ensure_eq!(
-            F::zero(),
-            (((witness[6][self.row + 1].double() + this[11]).double() + this[12]).double()
-                + this[13])
-                .double()
-                + this[14]
-                - this[6],
-            "endomul: wrong eq 11"
-        );
+        let evals: [ProofEvaluations<F>; 2] = [
+            ProofEvaluations {
+                w: this,
+                z: F::zero(),
+                s: array_init(|_| F::zero()),
+                lookup: None,
+                generic_selector: F::zero(),
+                poseidon_selector: F::zero(),
+            },
+            ProofEvaluations {
+                w: next,
+                z: F::zero(),
+                s: array_init(|_| F::zero()),
+                lookup: None,
+                generic_selector: F::zero(),
+                poseidon_selector: F::zero(),
+            },
+        ];
```

Comment:
perhaps we could use a function to initialize evals quickly if we're going to repeat this pattern across all the verify functions

Reply:
A `dummy` function (or perhaps an implementation of `Default`) would be good here.